### ISSUE TITLE
ExecutionFailed: arraycopy negative length

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13141]
When running the following query:
```cypher
UNWIND (CASE WHEN true THEN COUNT { () } END) AS n0 MATCH ()-[*1{n1:0, n2:0}]-() RETURN 0
```

I encountered the following error:
```
Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (arraycopy: length -1 is negative)
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.

I encountered this issue when testing queries against the **Neo4j 5.6.0 community version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
UNWIND (CASE WHEN true THEN COUNT { () } END) AS n0 MATCH ()-[*1{n1:0, n2:0}]-() RETURN 0
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (arraycopy: length -1 is negative)`.

**Comment by loveleif** [comment]
Thanks for the report @DominicWuest, much appreciated! I agree, this do look like a bug, I've created a ticket in our internal bug tracker for further investigation ([link](https://trello.com/c/JPFI3f6C/4696-when-running-the-following-query)).

**Comment by Lojjs** [comment]
@DominicWuest This bug has been fixed in Neo4j 5.8.0, sorry for the later reply on this one.

Best regards Louise, Neo4j Cypher team
